src/NBitcoin/ChainBase.cs

```diff
@@ -52,41 +75,54 @@ public ChainedBlock SetTip(ChainBase otherChain)
             return this.SetTip(otherChain.Tip);
         }
 
+        /// <summary>
+        /// Sets the tip of this chain based upon another block header.
+        /// </summary>
+        /// <param name="header">The block header to set to tip.</param>
+        /// <returns>Whether the tip was set successfully.</returns>
         public bool SetTip(BlockHeader header)
         {
             ChainedBlock chainedHeader;
             return this.TrySetTip(header, out chainedHeader);
         }
 
+        /// <summary>
+        /// Attempts to set the tip of this chain based upon another block header.
+        /// </summary>
+        /// <param name="header">The block header to set to tip.</param>
+        /// <param name="chainedHeader">The newly chained block for the tip.</param>
+        /// <returns>Whether the tip was set successfully.</returns>
```

Comment:
it would be useful to know when such an operation can fail

Reply:
Ok sounds good.